app/assets/javascripts/app/views/publisher/uploader_view.js

```diff
@@ -71,41 +64,39 @@ app.views.PublisherUploader = Backbone.View.extend({
 
   // replace the first photo placeholder with the finished uploaded image and
   // add the id to the publishers form
-  _addFinishedPhoto: function(id, image) {
+  _addFinishedPhoto: function(id, photoId, image) {
     var publisher = this.publisher;
 
     // add form input element
     publisher.$(".content_creation form").append(
-      "<input type=\"hidden\", value=\""+id+"\" name=\"photos[]\" />"
+      "<input type=\"hidden\", value=\"" + photoId + "\" name=\"photos[]\" />"
     );
     // replace placeholder
-    var placeholder = publisher.photozoneEl.find("li.loading").first();
+    var placeholder = publisher.photozoneEl.find("li.loading#upload-" + id);
     placeholder
-      .removeClass("loading")
       .prepend(
-        "<div class=\"x\"></div>"+
-        "<div class=\"circle\"></div>"
-       )
-      .find("img").attr(
-      {
-        "src": image.thumb_medium.url,
-        "data-small": image.thumb_small.url,
-        "data-scaled": image.scaled_full.url,
-        "data-id": id
-      }).removeClass("ajax-loader");
-    placeholder
-      .find("div.progress").remove();
+        "<div class=\"x\"></div>" +
+        "<div class=\"circle\"></div>" +
+        "<img src=\"" + image.url + "\" data-id=\"" + photoId + "\" alt=\"\" class=\"hidden\" />"
+      )
+      .removeClass("loading");
+    placeholder.find("div.progress").remove();
+    placeholder.find("img").on("load", function(ev) {
+      $(ev.target).removeClass("hidden");
+      placeholder.find(".spinner").remove();
+    });
+    this.$el.removeClass("loading");
```

Comment:
This line shouldn't be here, because the `loading` class on the uploader-button should only be removed after all uploads are finished, which is done below (I'm not sure if this class is used for anything at the moment, but if it's used, then it should only be removed after all uploads are done).

Reply:
Nice catch, I forgot it, it's duplicate it's also on the line below. Corrected.

Reply:
Actually, it should have been here :p it's not on the uploader-button, it's on the `li` containing the image. I'm putting it back.

Reply:
No, `this.$el` is the upload button, the `loading` class on the `li` element is removed here:

https://github.com/diaspora/diaspora/blob/522629ce40c29b4a23b03d7da45bc15bb93576c8/app/assets/javascripts/app/views/publisher/uploader_view.js#L82

Reply:
Hm this is weird because without that line [this test](https://travis-ci.org/diaspora/diaspora/jobs/393373631#L1877) is failing.

Reply:
But with this line the test is failing too.

I had a closer look at this now and the tests are probably broken since you fixed the first point of my first review (https://github.com/diaspora/diaspora/pull/7655#pullrequestreview-83994653), because since then the `onComplete` callback also uses the `id` to select the `li` element:

https://github.com/diaspora/diaspora/blob/522629ce40c29b4a23b03d7da45bc15bb93576c8/app/assets/javascripts/app/views/publisher/uploader_view.js#L75

And the `id="upload-x"` is missing here:

https://github.com/diaspora/diaspora/blob/522629ce40c29b4a23b03d7da45bc15bb93576c8/spec/javascripts/app/views/publisher_view_spec.js#L579

and you have `null` as `id` parameter here:

https://github.com/diaspora/diaspora/blob/522629ce40c29b4a23b03d7da45bc15bb93576c8/spec/javascripts/app/views/publisher_view_spec.js#L582

That's why it can't remove the `loading` class (and do everything else) from the `li`, because it can't select it. Hopefully that helps you with fixing the tests.